### DISCUSSION TITLE
Upgrading from 1.1.3 til 1.1.4 (and higher) breaks getOrElse: List<String> turns into List<Serializable>

### DISCUSSION THREAD

**Discussion by fatso83** [arrow-kt/arrow#3342] [Q&A] [answered]
I tried upgrading to 1.2.1, but got `Type mismatch: inferred type is List<Serializable> but List<String> was expected` on some existing code. I tried every older version until I found the change happened in 1.1.4. This is the code in question:
![image](https://github.com/arrow-kt/arrow/assets/618076/2ac24db8-b049-4ec3-9915-0faf1f4b8436)

It says the function `getContactPoints` returns `List<Serializable>`, which I cannot understand. `ContactPoint::contact` returns a string. Map returns a bog standard kotlin.collections List. If I remove the `getOrElse` I see that the object is ` Either<Throwable, List<String>>`, which is perfect. Somehow `getOrElse` changes String to Serializable. In the end I just slapped a `as List<String>` on it, but not sure why this is happening

I am not sure what happened in 1.1.4, as I could not see anything obvious [in the changelog](https://github.com/arrow-kt/arrow/releases/tag/1.1.4).

**Comment by kyay10** [comment, marked as answer]
I'm not sure how you expect this to work. What's happening is the compiler is merging `List<String>` and `List<Throwable>` into `List<Serializable>`, which, while mildly confusing, is valid enough. If `Serializable` didn't exist, then it'd just be `List<Any>`, and the same issue would arise. Did you perhaps change from `String` to `Throwable` somewhere?
